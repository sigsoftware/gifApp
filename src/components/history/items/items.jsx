import React from 'react';
import "./items.css";
import PropTypes from "prop-types";

export const Items = ({ item, term, setTerm }) => {
    return (
        <>
            <li className='item' onClick={() => setTerm(String(item))}><button>{item}</button></li>
        </>
    )
}

Items.defaultProps = {
    item: [{}],
    term: "",
}

Items.propTypes = {
    item: PropTypes.any.isRequired,
    term: PropTypes.string.isRequired,
}
