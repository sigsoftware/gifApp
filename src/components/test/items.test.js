
import { Items } from "../history/items/items";
import { shallow } from "enzyme";

describe("Pruebas al componente Items", () => {

    let wrapper;

    let item = "Final Fantasy";

    beforeEach(() => {
        wrapper = shallow(<Items item={item} />)
    })

    test("Renderizar la pagina correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    })

    test("Controlar si la clase de la etiqueta o elemento li tiene la clase items", () => {
        let divLi = wrapper.find('li').props();
        expect(divLi?.className).toBe('item');
    })

    test("Comprobar el contenido del boton de items", () => {
        let buttonText = wrapper.find('button').text().trim();
        expect(buttonText).toBe(item);
    })

    test("Simular el evento onClick", async() =>{
        let divLi = wrapper.find('li');
        wrapper.simulate("click");
        
    })
})