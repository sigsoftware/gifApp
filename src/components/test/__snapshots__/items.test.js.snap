// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente Items Renderizar la pagina correctamente 1`] = `
<Fragment>
  <li
    className="item"
    onClick={[Function]}
  >
    <button>
      Final Fantasy
    </button>
  </li>
</Fragment>
`;
